### PR TITLE
Check if config map created with leader-election

### PR DESCRIPTION
**Pull request by LorcanMcVeigh**
### Proposed changes
If the cli argument value is not a valid value for a resource name  and/or if the IC can't create a config map, the IC exits with an error 

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/master/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto master
- [x] I will ensure my PR is targeting the master branch and pulling from my branch from my own fork

### CODE REVIEW

File: cmd/nginx-ingress/main.go [hunk 1]
```diff
@@ -445,6 +451,16 @@ func getSocketClient(sockPath string) *http.Client {
 	}
 }
 
+// validateLockName validates the name of a resource
+// It returns an error if the resource name is invalid
+func validateLockName(lock string) error {
+	lst := validation.IsDNS1123Subdomain(lock)
```
Comment (Dean-Coakley) [inline cmd/nginx-ingress/main.go]: Lets use a more appropriate variable name. 
```suggestion
	allErrs := validation.IsDNS1123Subdomain(lock)
```

File: cmd/nginx-ingress/main.go [hunk 2]
```diff
@@ -445,6 +451,16 @@ func getSocketClient(sockPath string) *http.Client {
...
+	if len(lst) > 0 {
+		return fmt.Errorf("invalid resource name: %v", lock)
```
Comment (Dean-Coakley) [inline cmd/nginx-ingress/main.go]: Here we lose all context of the error returned from: `validation.IsDNS1123Subdomain(lock)`

Maybe something like:
```suggestion
		return fmt.Errorf("invalid resource name %v: %v", lock, allErrs)
```

Can work? As allErrs is a slice of strings.

File: pkg/apis/configuration/validation/validation.go [hunk 3]
```diff
@@ -323,8 +323,6 @@ func isValidHeaderValue(s string) []string {
 	return nil
 }
 
-// validateSecretName checks if a secret name is valid.
```
Comment (Dean-Coakley) [inline pkg/apis/configuration/validation/validation.go]: This doesn't relate to the PR. Lets revert these changes?

File: cmd/nginx-ingress/main.go [hunk 4]
```diff
@@ -445,6 +451,16 @@ func getSocketClient(sockPath string) *http.Client {
 	}
 }
 
+// validateLockName validates the name of a resource
+// It returns an error if the resource name is invalid
+func validateLockName(lock string) error {
```
Comment (Rulox) [inline cmd/nginx-ingress/main.go]: Maybe `validateResourceName` is a better function name ? That way we can re-use the validation function for other resources.

Comment (Dean-Coakley) [reply cmd/nginx-ingress/main.go]: @Rulox Do you suggest leaving `validateResourceName` in main.go and move to validate.go if/when we want to reuse it?

Comment (Rulox) [reply cmd/nginx-ingress/main.go]: Do you mean `validation.go` @Dean-Coakley ? I think main.go is better for this method.
 `validation.go` belongs to `pkg/apis/configuration/validation` This is, validation for CRDs (VS/VSR). 

Maybe we can also create another `.go` file for the validation of things that are not related with VS/VSR (or use `utils.go`). 

PS: there are more validation functions here in main.go so I don't think is a terrible idea to keep it here.
